### ISSUE TITLE
getTagByName will fail for tag without tag message

### ISSUE THREAD

**Issue by prsyahmi** [nodegit/nodegit#1311]
When calling getTagByName on a tag without tag message, it will throw an error:
`Error: the requested type does not match the type in the ODB errno: -3`

**Comment by rcjsuen** [comment]
What information are you trying to get out of the `Tag` object, @prsyahmi? The lightweight tag itself has no information so even though you have a `Tag` object, I don't think you'll be able to get anything useful out of it.

**Comment by prsyahmi** [comment]
I tried to get the `Commit` object/commit oid out of `Tag` (`Tag.targetId`). To reproduce:
1. Tag any commit with no message: `git tag test-tag HEAD`
2. On opened repository object, call getTagByName: `repo.getTagByName('test-tag')`

**Comment by rcjsuen** [comment]
@prsyahmi If you are okay with just retrieving the tag's underlying `Commit` object, then use `repository.getReferenceCommit("refs/tags/my-tag")`.

**Comment by chrisdostert** [comment]
First of all, this library is great so thank you all. @rcjsuen I also ran into this and it was hard to discover this behavior.  Is there hesitation to add "lightweight" tag support to repo.getTagByName?

**Comment by rcjsuen** [comment]
@chrisdostert It's not possible to add lightweight tags because lightweight tags is "just a name" for a commit. There is no physical `Tag` object in the Git object database associated with a lightweight tag so the `getTagByName` function can't return anything meaningful.

**Comment by chrisdostert** [comment]
@rcjsuen right, but I think from an intuitive API design perspective, abstracting that underlying implementation detail is appropriate. The tag returned would just have null fields for those which didn't apply (id, tagger, message, etc...).  `git tag` command for example returns both "annotated" and "lightweight" tags; which i'd expect is most people coming to use the library will be familiar with.
